source/codegen/templates/service.h.mako

```diff
@@ -37,26 +43,45 @@ resource_repository_ptr = f"std::shared_ptr<{resource_repository_type}>"
 #include <server/session_resource_repository.h>
 #include <server/shared_library.h>
 #include <server/exceptions.h>
+%if has_async_functions:
+#include <server/async_method_context.h>
+%endif
 
 #include "${config["module_name"]}_library_interface.h"
 
 namespace ${config["namespace_component"]}_grpc {
 
-class ${service_class_prefix}Service final : public ${service_class_prefix}::Service {
+class ${service_class_prefix}Service final : public ${base_class_name} {
 public:
   using ResourceRepositorySharedPtr = ${resource_repository_ptr};
 
   ${service_class_prefix}Service(${service_class_prefix}LibraryInterface* library, ResourceRepositorySharedPtr session_repository);
   virtual ~${service_class_prefix}Service();
   
+%if has_async_functions:
+  void register_async_functions(::grpc::ServerCompletionQueue* completion_queue);
+%endif
+% for function in async_functions.keys():
```

Comment:
TODO: shouldn't just copy/paste this. Maybe I can just put the `using` statements next to the method.